### PR TITLE
feat(DatePicker): update date library package

### LINKED ISSUE #742: [Feature]: Support other Calendar Systems
### Describe the feature

https://react-spectrum.adobe.com/blog/date-and-time-pickers-for-all.html#calendar-systems

React-aria uses `Provider` (context) for locale in Dates but in `radix-vue` it's not clear how to change Calendar Systems and locale correctly

--- 

For example PersianCalendar System

### RadixVue

https://stackblitz.com/edit/plbydl

- Persian date segments are `YYYY/MM/DD` ✅  but `v-for` segments are `DD/MM/YYYY` ❌
if I reverse segments array keyboard arrows will not work good with it

- If I select a date cell when my Calendar System is Persian, my DatePicker will Teleport time from 1402 to 2024 or 3266 and ...

### MeltUI

https://stackblitz.com/edit/github-yzqybw-guqhpr

- Persian date segments are correct `YYYY/MM/DD` ✅ 

### Additional information

- [ ] I intend to submit a PR for this feature.
- [ ] I have already implemented and/or tested this feature.

### PR DESCRIPTION
**Pull request by epr3**
Closes #742.

Hello,

I forked and refactored Adobe's `@internationalized/date` package and converted all ES6 classes into plain objects and converted the original class methods into functions. All functions return new calendar objects and are immutable.

Then, I've refactored the whole implementation of the `Calendar`, `DateField` and `DatePicker` components to use this library and removed some redundant functions, updated the docs and added localised stories for some of the calendars in Histoire. 

It should fix all errors that were caused by Vue changing the `this` context from the `DateValue` class to it's own `Proxy` class (plus any date jumps to the Gregorian Calendar and back causing dates to desync).

This is the library: https://www.npmjs.com/package/flat-internationalized-date

If you guys want, I can relinquish the package ownership to the `radix-vue` org in Github and continue maintenance under `radix-vue`.

I'm looking forward to any feedback and improvements you suggest.

Thanks!

CC-ing @sadeghbarati for the Persian Calendar fixes.

### CODE REVIEW

**Comment by sadeghbarati** [comment]
:heart:

a Question,

How to use another Calendar System without setting `year`, `month` and `day` manually

#### before:

```ts
const persian = createCalendarDate({
  calendar: CALENDAR.PERSIAN,
  year: 1403,
  month: 12,
  day: 1,
})
```

#### after: 

```ts
const persian = createCalendarDate({
  calendar: CALENDAR.PERSIAN,
})
```

Automatically use `today` as the date with related Calendar System

**Comment by epr3** [comment]
@sadeghbarati You could try this approach for getting today's date in a different calendar system.
`
import { today, getLocalTimeZone, toCalendar, CALENDAR } from 'flat-internationalized-date'

const persian = toCalendar(today(getLocalTimeZone()), CALENDAR.PERSIAN)
`

**Review by zernonia — COMMENTED** [review]
Phenomenal  work @epr3 ! Will have a look asap! 💚

**Comment by MellKam** [comment]
So, why do we need to copy `@internationalized/date`? Just to use objects instead of classes? This would require constant maintenance and, in general, a lot of work. Isn't it simple to rely on existing popular package maintained by giant company? Is is really impossible to fix this context lost issue without rewriting whole library?

**Comment by epr3** [comment]
I'm open to any ideas getting `@internationalized/date` to work without the classes losing context. 

As for the maintenance burden, I can maintain the fork of the library, keeping feature parity with the original library (which I think it shouldn't be that hard since the last notable code change was made ~3-8 months ago).

**Comment by sadeghbarati** [comment]
@MellKam 

https://github.com/tc39/proposal-class-fields/issues/106#issuecomment-401955418
https://javascript.info/proxy#proxy-limitations

**Comment by MellKam** [comment]
@sadeghbarati I looked at the `@internationalized/date` codebase and didn't find any private fields in it (except for those `#type` fields that are used only to distinguish the shape of objects). 

https://github.com/adobe/react-spectrum/blob/9d37e2420a229f3f5f7267f4c63d1549543e8626/packages/%40internationalized/date/src/CalendarDate.ts#L42-L46

**Comment by MellKam** [comment]
Here is [playground](https://play.vuejs.org/#eNp9VEtv2zAM/iuEL0kAx273OmRJ0K0LsA57Ye1t2kG1mVidLRmSnMcC//dRUuykRZtcwufHj6SpffShrpN1g9EkmppMi9qCQdvUUHK5mrHIGhbNmRRVrbSFPWhcxrDhNiughaVWFQwoe3AScc1LlDnXn7jFGDKN9N/ZYrCqk7t8Fl0JaVFLboWSvBT/ME9zSmIRk0xmShoLToeZqz6UuHlUY/jq4uJtDJcxvBuN3ncJWVclJLn8ZM3LBpPOk4gcpRVLgdqlMem7GnbuGIYjmM1hzyT48iGd8I4tnMA+7bTHCW7PrB0xOU3DlGmmpFis6pKySAOYGiwxs+BkgKusoA0grWCIayJ6JON+oUkaRVfvpm+GGIaMxHK9QgvcwOe7b19vPfqixMqhBVpE6oB3HNezoCGudSsB8GyJr6rdxg58AdbjpdLEl7BALeH3oEZtBJeDGAYrjSuld04UMj8YMweQOanAe40bJ903eV4IYwd/WNQjTzzZAH1qDgPD/FB0Npv1jQSmPVeA/d75oG0P5NPAPkw+DUhBKS7nFOxXu0P3nbbjTq+UtPThHw0535E6TSnH7fZkn1FMt0NrWopV8mCUpAPz2yOqqqpFifqHJ0D3Nen2yiJelmrzxduspq+qs2cFZn+fsT+YrbOx6KdGg3pNR9P7wvqDe3H7Hbck985K5U1J0Wecv9CosnEcQ9jHRuZE+yTOs73xhy/k6s4sthal6ZpyRF1k6+NZRO/E9ZnWj3RfJ298Hl0MTTE8LOOK10/mGByPQV58SlwDhbW1maQpmioxRXrm2TkUb/8DLlGz2g==) where I tried to change the calendar of date, and it worked pretty well. Seems like all objects in `@internationalized/date` are immutable, so this shouldn't cause any issues.

**Comment by epr3** [comment]
@MellKam Got it, but I believe the `DateValue` `toString` method does not compute/reactively update even using the playground.

I patched `CalendarDefault.story.vue` file with some dummy values and get errors like in the video. (the functions that are undefined differ, sometimes it's `getYearsInEra`, sometimes it's `getJulianDay`, etc.)

https://github.com/radix-vue/radix-vue/assets/17836403/1251ca32-5e58-454e-a6af-f75c99989ac1

Would the errors be related to bugs in how Histoire does its rendering?

Here is the patched story:
```
<script setup lang="ts">
import { Icon } from '@iconify/vue'
import { CalendarCell, CalendarCellTrigger, CalendarGrid, CalendarGridBody, CalendarGridHead, CalendarGridRow, CalendarHeadCell, CalendarHeader, CalendarHeading, CalendarNext, CalendarPrev, CalendarRoot } from '../'
import { type DateValue, GregorianCalendar, PersianCalendar, getLocalTimeZone, toCalendar, today } from '@internationalized/date'
import { type Ref, onMounted, ref } from 'vue'

const locale = ref('en')
const day = ref(toCalendar(today(getLocalTimeZone()), new GregorianCalendar())) as Ref<DateValue>

onMounted(() => {
  setTimeout(() => {
    locale.value = 'fa'
    day.value = toCalendar(today(getLocalTimeZone()), new PersianCalendar())
  }, 5000)
})
</script>

<template>
  <Story title="Calendar/Default" :layout="{ type: 'single' }">
    <Variant title="default">
      {{ day.toString() }}
      <CalendarRoot
        v-slot="{ weekDays, grid }"
        v-model="day"
        class="mt-6 rounded-xl border border-black bg-white p-4 shadow-md"
      >
        <CalendarHeader class="flex items-center justify-between">
          <CalendarPrev
            class="inline-flex items-center cursor-pointer text-black justify-center rounded-[9px] bg-transparent w-10 h-10 hover:bg-black hover:text-white active:scale-98 active:transition-all focus:shadow-[0_0_0_2px] focus:shadow-black"
          >
            <Icon icon="radix-icons:chevron-left" class="w-6 h-6" />
          </CalendarPrev>

          <CalendarHeading class="text-[15px] text-black font-medium" />

          <CalendarNext
            class="inline-flex items-center cursor-pointer text-black justify-center rounded-[9px] bg-transparent w-10 h-10 hover:bg-black hover:text-white active:scale-98 active:transition-all focus:shadow-[0_0_0_2px] focus:shadow-black"
          >
            <Icon icon="radix-icons:chevron-right" class="w-6 h-6" />
          </CalendarNext>
        </CalendarHeader>

        <div
          class="flex flex-col space-y-4 pt-4 sm:flex-row sm:space-x-4 sm:space-y-0"
        >
          <CalendarGrid v-for="month in grid" :key="month.value.toString()" class="w-full border-collapse select-none space-y-1">
            <CalendarGridHead>
              <CalendarGridRow class="mb-1 grid w-full grid-cols-7">
                <CalendarHeadCell
                  v-for="day in weekDays" :key="day"
                  class="rounded-md text-xs !font-normal text-black"
                >
                  {{ day }}
                </CalendarHeadCell>
              </CalendarGridRow>
            </CalendarGridHead>
            <CalendarGridBody class="grid">
              <CalendarGridRow v-for="(weekDates, index) in month.rows" :key="`weekDate-${index}`" class="grid grid-cols-7">
                <CalendarCell
                  v-for="weekDate in weekDates"
                  :key="weekDate.toString()"
                  :date="weekDate"
                  class="relative text-center text-sm"
                >
                  <CalendarCellTrigger
                    :day="weekDate"
                    :month="month.value"
                    class="relative flex items-center justify-center whitespace-nowrap rounded-lg border border-transparent bg-transparent text-sm font-normal text-black w-8 h-8 outline-none focus:shadow-[0_0_0_2px] focus:shadow-black hover:border-black data-[selected]:bg-black data-[selected]:font-medium data-[disabled]:text-black/30 data-[selected]:text-white data-[unavailable]:text-black/30 data-[unavailable]:line-through before:absolute before:top-[5px] before:hidden before:rounded-full before:w-1 before:h-1 before:bg-white data-[today]:before:block data-[today]:before:bg-grass9 data-[selected]:before:bg-white"
                  />
                </CalendarCell>
              </CalendarGridRow>
            </CalendarGridBody>
          </CalendarGrid>
        </div>
      </CalendarRoot>
    </Variant>
  </Story>
</template>
```

**Comment by sadeghbarati** [comment]
- [Eduard story code as Stackblitz](https://stackblitz.com/edit/plbydl-x6jkcp)
- [Time jump with es6 classes](https://stackblitz.com/edit/plbydl-3infpv)

**Comment by MellKam** [comment]
I see, thanks for reproductions ❤️ 

`this` keyword doesn't mean anything bad. Vue's reactivity system can work with it. The only problem is with private fields. The ones that start with `#`. Here is [playground example](https://play.vuejs.org/#eNqNk19P2zAQwL/KzXugFTTdtD2xFG1DPLCHDQ0eLaEQrq3BsSP/aYuifHfOdt2mCFVIbdXc/S753Z3TsV9tW6w8snNW2tqI1oFF51uQlVrMOHOWswuuRNNq46ADg3PoYW50AydUdsIVV7WsrIVL7ZVDAx1XAK0Rq8oh3NchCjP4EkCABTqIodE4gUB3dN4ocEthi4T/CIk+/ISvULXBBoclA/b0NNP0ySo36ekHRp93JlTwAZXE55sfE0nkWxGt7Pb+ZDALcxspXOcpjcbjzGxHlWX36GEXsaKcph3RRujCYdNKIugKoFx+vdii59B1+dFFarvvyykBEXzwzmkFP2sp6mdacSYH/dHOY7acJjgVfppM4HoOL9qDJ/kmsHEEMbQWUsahVgrQGG1gMtnXBb1tQ1ktah52/wHbNwXHpJNAOR3MiZ3Riaa5z8WieLJa0bGPqwxTaFoh0fxrnaC9cEZ6acmcVVLq9Z8Yc8bjWY7XS6yf34k/2U2IcXZj0KJZIWe7nKsMDSmlr27/4ob+75KNfvSS6CPJ/2i19MExYb+9eiTtARdtr+MLK9Tizl5tHCqbmwqi8aBGnjN6iS+PtL7X/VZ8zwec9a/Rs23v)

I don't know what exactly is wrong with the current Calendar implementation, but for sure this is not `@internationalized/date` library and vue's reactivity. Here is [my attemps to break the reactivity](https://play.vuejs.org/#eNqNVX9v2zYQ/SoHAYVtTLKdtdkGLw7StcGWotmKNRiwVcVAS7TFRCIFkvKPBv7ueyRtRU6cYP/YFO/u8d4d+e4+elvXw2XDo0l0ZjItakuG26amksnFNI2sSaPzVIqqVtrSPSl5rRppeR6T5nPa0lyrinpA6LVe71jJZc50sKXRcLTfcSel0c8dvF81XygtmNy7xGTVw3rB7UeVsfJGVPwfJbmz5mwT0yeuzUFUpjmzvD16l1gaXQhkqyWzQklWim88H+VwTKNUpjJT0lgCIE0dnf7D0Vhiu//4/P5gEJPkq6d5wzIAsbNRqCJqhg/Lq7rEafgiOjM1k+efeckzS7bg/lyfZaWkLRDp7N6zpbFMKpXzEo2AcxrRaIcUQNya6CIr0CsOnz5fcmkHND2n+2AjChSR8h7zKoePmAuuQTpEDC3TYErM0G831x9DipclrxzacMnKhoNawAvF6hTKt+Ow+v2jx/n6OIitKz6RpwIyqna92ZEhUJ4rDTKoOqk5fenVodW9mHoLX/WNWwqZ7zYzB5C5VcFnmq/catbkeSGM7X1F0fbIE88kQHe3QzV5vjt0Op06lkOsw6UVLYWQeJs60f09uZDtdsdlFMiELo0CcKf5cHfIG+5u6DbZffr2d75dibfb9j6cjTrXCJ/Gbkq3nKl8E/o8B0AyZ5UoNxMyGwP/pBG+2iuR22JCJ+PxcuU3Ci4WhQ07hd+ZsexuofGq80RUbMEnVAqJDJOFZrkAdbwFmilrcVO1C46pMGX/5MfXMf00Ho5fxfT6B/wNyC296fQkptNTb3pzMjyFCce9Cv3PhQEZJDov+drv4FkuZCKQtQm7icGFtN522xiUfpPgGlukMqEMv1x7W83yXMhFYlUNPt+P6wCXrPjsTqAgriqmUsoW8JoQQxNxFDM8D36V+pYos37iCN4b47oPNzQWjdhVPIqhh8hkLhbDW6MkRNPXH/dGVbUouf7Dtx+aOdm/wDRiZalWH/ye1Q2P9/tZwbO7I/u3Zu320uiT5obrJbSqtYWHGsyXn3/na6xbI6SiKeH9gvFPblTZuByD2y/oOtLu+Plsr7w8oxg35nKNups9KZeo89x6/zSCnr97gfpDuq+Hb3wcKooqBvlPKlY/qmMwHII8q+COQGFtbSajETfV0BSjZ31bfu5OrxM/h47Ed6xtxIVAy3EHR88FHTp0aHbH3v8dsO+R7l9Op9rZ+gynHm7nwwSzHKqc8zle7rWbGGctznl/8PJYysVyL8VuadWHBq9Evndv1CuS5Q9a2LFC+S0feKV6BIGe/w0BMVfyUrMjIAf24zCzBnrzdCj0/41JDEhIeqvxSn3sAS7yAuy1U9QAPBjORVn2x4OO4l9kpcju/FANdXMo6Erf6++EBH1HJ7RFSDsj4MKSzqDwIX4WYlp4/2OzYQfUTofAajcdAtuj6k5QmZrnTuR9xJfD490r/BoeSEe8M1UqPaHZflrvvlcFlPWJkm3/A+EkV08=), but I couldn't do that. I will try to figure out the source code of Calendar when I have more free time. (it seems like a lot of code to read 😅)

**Comment by epr3** [comment]
@MellKam if it's of any help, I've set up a separate branch only with the code fixes from this branch and using `@internationalized/date` here: https://github.com/epr3/radix-vue/tree/742-calendar-systems-int-date

**Comment by MellKam** [comment]
@epr3 Thanks! That will help

**Comment by MellKam** [comment]
I think the problem lies in this function. You need to `.copy()` the dates before you return them. For me it fixes the story with controlled value.

https://github.com/radix-vue/radix-vue/blob/c2e5c1ca7f55880f5d42769b18cced3b5a8e86af/packages/radix-vue/src/shared/date/comparators.ts#L33-L58

**Comment by MellKam** [comment]
https://github.com/MellKam/radix-vue/commit/feb88013046cceed4c3d05bab4cf7319fed64733

**Comment by epr3** [comment]
@MellKam I pulled in your repo and set it up, looks like Histoire still outputs errors:

https://github.com/radix-vue/radix-vue/assets/17836403/aefb8bf3-931e-412f-9c63-13973cadd24a

But on [Stackblitz](https://stackblitz.com/edit/kmbghl-rbaue6?file=src%2FApp.vue), it is working properly. 👍🏻

**Comment by MellKam** [comment]
It's strange because it doesn't work for me on the histoire page either, but it works when I open it in a separate window. Maybe it is some problem with the histoire cache

**Comment by epr3** [comment]
Different browser, incognito window:

https://github.com/radix-vue/radix-vue/assets/17836403/9923f21e-4edc-4314-b38e-d8a1301f0dbd

Removed `node_modules`, ran a `pnpm i --force`, then opened the story in an incognito window:

https://github.com/radix-vue/radix-vue/assets/17836403/903a12e2-da15-454d-a257-ae86f28083bc

Am I doing something wrong?

**Comment by MellKam** [comment]
Wait, now it's a different error. Let me check it

**Comment by MellKam** [comment]
All error that we see are related to not found the method in object prototype. For some reason the calendar object can lose it's prototype methods. I dont know at which point this happens. Tried to reproduce it in playground, but without success. Maybe this is actually vue's bug or I just dont see some stupid mistace. Will come back to it this evening.

**Comment by epr3** [comment]
One culprit that I could identify (other than Vue's reactivity system) which might cause the `DateValue` prototype to be reset to a plain object might also be `useVModel` which does some clones internally and by default it uses `JSON.parse(JSON.stringify(value))` to do the clones.

Reference: https://github.com/vueuse/vueuse/blob/1558cd2b5b019abc1feda6d702caa1053a182903/packages/core/useVModel/index.ts#L113

**Comment by MellKam** [comment]
@epr3 That's definitely it! Here is updated version of my [previous playground](https://play.vuejs.org/#eNqNVmFv2zYQ/SsHAYVtTLKTpdkGLQnStcGWotmCNSiwRcEgS2eLiSQKJGXHDfzf90jaipI43b7YJO/u3b078k4PwbumGS9aDuLgSGdKNIY0m7ahMq3nx0lgdBKcJLWoGqkMPZCsL2RbG85DUjyjNc2UrGgAhEGn9T4tuc5T5WVJMJ5sT6ynJPi5h/er4rlUIq23KiEZ+bies/kks7S8EhX/LWu20jxdhXTJSj+xyhSnhjvXm8CS4FQgWlWnRsg6LcVXzic5FJMgqZM6k7U2BEA6tnSGj66xxPHwuf/haBRSzcuXcUMyArGjic8icoaN4aop4Q07oiPdpPXJZy45M2QKdn5dlJWsTQFLK3eaHY1FVMmcSxQCyklAkw2SB7FrotOsQK0YOkNecG1GdHxCD15G5Cki5C3meQ4dMROsQNpbjE2qwJRSTb9dXXzyIZ6VXFm08SItWwY1j+eT1UuUK8fT7A93unP5sRBrm3wiRwVkZGNrsyFDoDyTCmSQdZIzuh40vtSDkAZzl/WVXYo63xxmFiCzq4Knipd2NW3zvBDaDG6QtC1y7Jh46P6xzybnG6fHx8eW5Rhrf2lFR8EH3oVO9PBA1mS93nCZeDK+ShMP3Cs+1C3yiu0NXUebrSt/b29TvF539+Fo0rtG2GqzKu1yKvOVr/MMANEsrUS5ikmvNPSjVrhsL0Vuipj29/YWS3dQsJgXxp8U7mSaZndzhVedR6JK5xxTKWpEGM1VmgtQx1ugqTQGN1VZ45AKXQ73fzwI6ae98d6bkA5+wN+I7NKJDvdDOjx0orf740OI4O6Nr38uNMgg0FnJ9+4Ez3JeRwJRa38aaVxI42S3rUbqVxGusUEoMWX4ZeVkTZrnop5HRjbg8/1e4+GiJU/vBBJis6IrKU0BrZhSFBGuUs2516vk10jq+xeK4L3StvpQQ2FRiE3GgxD9EJHMxHx8q2WNpunyj3sjq0aUrP5w5UfPjLcvMAnSspTLj+7MqJbD7XlWcHa34/xW39uzJLhUrFkt0Ks6mX+oXnz2+Xe+x7oTolW0JbS/IfyTtSxbG6NX+wVVR9g9PRftuWvPSMaVPrtH3vWWlA3Uaq6dfhKgn7//BvXHcA/Gb50dMoos+vYfVWnzLI9e8BTk1Q5uCRTGNDqeTFhXY11MXtXt+Nk7fR+5ObTDviftLE6xbTVPMqlecfpEoUezP/b+74D9gHC/2D7VzdZXOA36YxTuv1zYSdEbfM+i2k67RslGo4XnPMMzv7S7oweMIBg7v3E/hJOhfbTekCthOrszbKzdoG1sMPGj/SCm60UP5Mah9KetYcB0EQ9dQOjZPYjQOeuMZMnjUs4xagz7cfRfozYXi+14sUsjP7Z4+fUH23dclwVQ1997Uudi5LrvMwjc47/QFPV5fabSHSBP5Lthpi166MtBN/wnJDEiUdM7hc7jWfZxERdgL+yU8MCj8UyU5XBv1Jtip1kpsjv3oeDy61Bw04ZupsQk6DvapzVMurkHlTTqDT9n4uY7JqDT3zXvNkDdxPOsNhPPs905sQids+HcDi5ncf3Uve0sN/7R9wZSJkupYppuv0A2+2WBafGiO6//BRGuoyQ=) but I changed `defineModel` to `useVModel` from  `@vueuse/core`

**Comment by epr3** [comment]
I'm overwriting the `useVModel` values in `CalendarRoot` with this (to get over `JSON.parse`):
```const modelValue = useVModel(props, 'modelValue', emits, {
  defaultValue: props.defaultValue ?? undefined,
  passive: (props.modelValue === undefined) as false,
  clone: value => Array.isArray(value) ? value.map(item => item.copy()) : (value?.copy() as DateValue),
}) as Ref<DateValue | DateValue[] | undefined>

const defaultDate = getDefaultDate({
  defaultPlaceholder: props.placeholder,
  defaultValue: modelValue.value,
})

const placeholder = useVModel(props, 'placeholder', emits, {
  defaultValue: props.defaultPlaceholder ?? defaultDate.copy(),
  passive: (props.placeholder === undefined) as false,
  clone: value => value?.copy(),
}) as Ref<DateValue>
```
But it seems that `DateValue` loses its prototype still:

https://github.com/radix-vue/radix-vue/assets/17836403/4b3bc787-277b-4b26-bf60-98fca3c32f12

**Comment by epr3** [comment]
@MellKam I think I finally figured out the what might be the issue.

I've pulled in your [playground code](https://play.vuejs.org/#eNqNVmFv2zYQ/SsHAYVtTLKTpdkGLQnStcGWotmCNSiwRcEgS2eLiSQKJGXHDfzf90jaipI43b7YJO/u3b078k4PwbumGS9aDuLgSGdKNIY0m7ahMq3nx0lgdBKcJLWoGqkMPZCsL2RbG85DUjyjNc2UrGgAhEGn9T4tuc5T5WVJMJ5sT6ynJPi5h/er4rlUIq23KiEZ+bies/kks7S8EhX/LWu20jxdhXTJSj+xyhSnhjvXm8CS4FQgWlWnRsg6LcVXzic5FJMgqZM6k7U2BEA6tnSGj66xxPHwuf/haBRSzcuXcUMyArGjic8icoaN4aop4Q07oiPdpPXJZy45M2QKdn5dlJWsTQFLK3eaHY1FVMmcSxQCyklAkw2SB7FrotOsQK0YOkNecG1GdHxCD15G5Cki5C3meQ4dMROsQNpbjE2qwJRSTb9dXXzyIZ6VXFm08SItWwY1j+eT1UuUK8fT7A93unP5sRBrm3wiRwVkZGNrsyFDoDyTCmSQdZIzuh40vtSDkAZzl/WVXYo63xxmFiCzq4Knipd2NW3zvBDaDG6QtC1y7Jh46P6xzybnG6fHx8eW5Rhrf2lFR8EH3oVO9PBA1mS93nCZeDK+ShMP3Cs+1C3yiu0NXUebrSt/b29TvF539+Fo0rtG2GqzKu1yKvOVr/MMANEsrUS5ikmvNPSjVrhsL0Vuipj29/YWS3dQsJgXxp8U7mSaZndzhVedR6JK5xxTKWpEGM1VmgtQx1ugqTQGN1VZ45AKXQ73fzwI6ae98d6bkA5+wN+I7NKJDvdDOjx0orf740OI4O6Nr38uNMgg0FnJ9+4Ez3JeRwJRa38aaVxI42S3rUbqVxGusUEoMWX4ZeVkTZrnop5HRjbg8/1e4+GiJU/vBBJis6IrKU0BrZhSFBGuUs2516vk10jq+xeK4L3StvpQQ2FRiE3GgxD9EJHMxHx8q2WNpunyj3sjq0aUrP5w5UfPjLcvMAnSspTLj+7MqJbD7XlWcHa34/xW39uzJLhUrFkt0Ks6mX+oXnz2+Xe+x7oTolW0JbS/IfyTtSxbG6NX+wVVR9g9PRftuWvPSMaVPrtH3vWWlA3Uaq6dfhKgn7//BvXHcA/Gb50dMoos+vYfVWnzLI9e8BTk1Q5uCRTGNDqeTFhXY11MXtXt+Nk7fR+5ObTDviftLE6xbTVPMqlecfpEoUezP/b+74D9gHC/2D7VzdZXOA36YxTuv1zYSdEbfM+i2k67RslGo4XnPMMzv7S7oweMIBg7v3E/hJOhfbTekCthOrszbKzdoG1sMPGj/SCm60UP5Mah9KetYcB0EQ9dQOjZPYjQOeuMZMnjUs4xagz7cfRfozYXi+14sUsjP7Z4+fUH23dclwVQ1997Uudi5LrvMwjc47/QFPV5fabSHSBP5Lthpi166MtBN/wnJDEiUdM7hc7jWfZxERdgL+yU8MCj8UyU5XBv1Jtip1kpsjv3oeDy61Bw04ZupsQk6DvapzVMurkHlTTqDT9n4uY7JqDT3zXvNkDdxPOsNhPPs905sQids+HcDi5ncf3Uve0sN/7R9wZSJkupYppuv0A2+2WBafGiO6//BRGuoyQ=) into a **Histoire** story for debugging and I found out that while everything might work fine in a normal application, Histoire might do some shenanigans (like `JSON.parse(JSON.stringify(DateValue))` ) while passing data from the story iframe (or rendered story) back to the `v-model` variable, thus losing the prototype. Also, setting `{ iframe: false }` on the Story layout doesn't work to maybe circumvent some `iframe` `postMessage` serialization.

Video for reference:

https://github.com/radix-vue/radix-vue/assets/17836403/80b3f496-9468-4f87-9503-fda1fb20c698

In short, the issue won't happen in a normal `Vue` app, it's related to **Histoire** and **Histoire only**.

I'll revert the date library back to `@internationalized/date`, revert all the methods, tests, stories and docs and keep the other fixes on this branch to be merged.

I'll also dig through Histoire and see how it passes data between its `Vue` apps (app shell <-> story communication) and open an issue + PR there if I find anything amiss and I can fix.

Sorry for the delays in getting these fixes merged.

cc @sadeghbarati Unfortunately, until Adobe merges this [PR](https://github.com/adobe/react-spectrum/pull/6077), the `PersianCalendar` will output incorrect dates on some months.

**Comment by sadeghbarati** [comment]
@epr3 that OK ❤️ I'll patch it

**Comment by epr3** [comment]
@zernonia I've reverted the date library back to `@internationalized/date`. PR is ready for another review 🚀 

cc @MellKam

File: packages/radix-vue/src/DateField/DateFieldRoot.vue [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
-  type SupportedLocale,
   getDefaultDate,
   hasTime,
   isBefore,
 } from '@/shared/date'
 import { createContent, initializeSegmentValues, isSegmentNavigationKey, syncSegmentValues } from './utils'
 
 type DateFieldRootContext = {
-  locale: Ref<SupportedLocale>
+  locale: Ref<string>
```
Comment (zernonia) [inline packages/radix-vue/src/DateField/DateFieldRoot.vue:20]: @epr3  why change from the `SupportedLocale` type to `string` ya? This loses the ability to autcomplete the available locale right?

**Review by zernonia — COMMENTED** [review]
Just a quick question, and I'm thoroughly amazed by all the discussion, research and back-and-forth testing to nail down the issue here!

Really apprecate @epr3 @sadeghbarati  @MellKam !!! 💚💯

File: packages/radix-vue/src/DateField/DateFieldRoot.vue (hunk 1 above)
Comment (epr3) [reply packages/radix-vue/src/DateField/DateFieldRoot.vue:20]: @zernonia The `locale` prop is only used by the `DateFormatter` which is just a wrapper with some polyfills over [`Intl.DateTimeFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat), so just using `SupportedLocale` would not cover all the possible values that are accepted. e.g. `fa-IR`, `de-AT`, `de-DE`, `zh-Hans-CN`

`SupportedLocale` is used for the `DateField` type components to generate localized strings for the placeholders like `YYYY/MM/DD`.

Comment (zernonia) [reply packages/radix-vue/src/DateField/DateFieldRoot.vue:20]: Ahh ic! Thanks for the details!
